include/stl2/view/move.hpp

```diff
@@ -0,0 +1,83 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter 2016
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_DETAIL_VIEW_MOVE_HPP
+#define STL2_DETAIL_VIEW_MOVE_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/concepts/object.hpp>
+#include <stl2/detail/iterator/move_iterator.hpp>
+#include <stl2/detail/range/concepts.hpp>
+
+STL2_OPEN_NAMESPACE {
+	template <InputRange Rng>
+	class move_view {
```

Comment:
A range may or may not own its contents. Views are the subset of ranges that are `Semiregular` and have copy and assignment that are O(1) relative to the number of elements. That O(1) requirement implies that views either do not own the elements they denote, or that they own a constant number of elements.

> I was confused as the move_view view adaptor copies the Range it is adapting, and so I didn't saw it as a view.

`InputRange` is not the correct constraint for the class parameter: it's required to satisfy both `View` and `InputRange` so that copy is guaranteed to be O(1). [That constraint is corrected in a later commit](https://github.com/CaseyCarter/cmcstl2/commit/49a0a43641f34ba1eb8170a1670a00ee3c0a8975#diff-5d0d31fea97eb927a7b79f10f1c47095L21).